first_order_model/shrink_util.py

```diff
@@ -1015,40 +1042,62 @@ def compute_deletion(layer_graph,
         follow(following_layers, layer_graph, layer_graph[after_layer])
 
     deletions = {}
+
+    # In this if/else if we are finding the output features to remove from layer
     if isinstance(custom, list):
+        # Either its passed in as a list in some cases (like if this output gets added to another one)
+        # and we've already determined what to remove from the other output.
         deletions[layer] = copy.copy(custom)
     elif isinstance(custom, int):
-        # The sorting magic happens here
-        # Find the first convolution
+
+        # Or custom tells us how many features to remove, but we decide which ones to remove.
         if sort:
+            """
+            In general the approach here is as follows
+
+            We need to calculate the "importances" of a given feature, which we determine using the following layer's weighting for that feature.
+
+            1) We need to find the following layer (called first_conv)
+
+            2) Figure out which of its input features correspond to the output features in layer (called slices)
+
+            3) Calculate the importance for a given feature by summing up its weights in the following layer.
+
+            """
+            # In the sorting case we select by lowest batch norm.
             assert layer_graph[layer].type != 'bn', "We cant sort batchnorm layers"
 
-            # Get the layer which accepts this ones output as input.
+            # Get the layer which accepts this ones output as input. In general there is only one such layer.
+            # But with skip connections there may be multiple such layers as this output can be fed to a different layer.
+            # In that case, we just select the non-skip connection.
             depthwise_skipped_following_layers = []
             for after_layer in layer_graph[layer].after:
                 get_following_layers_skip_depthwise(depthwise_skipped_following_layers, layer_graph, layer_graph[after_layer])
 
+            # first_conv is the layer described above, which we will use to determine the importances of the features.
             first_conv = get_first_conv(layer_graph, depthwise_skipped_following_layers)
 
             assert first_conv is not None, "First conv is none"
 
-            # Find the relevant slices of input which matters if there is a concat in the way
+            # If output of layer gets concatenated with something (could be itself), we need to figure out which "slices" in first_conv's inputs are actually affected by layer's output.
```

Comment:
I'm not sure I fully follow this. An example would help

Reply:
Explained.